tree/dataframe/src/RDFHelpers.cxx

```diff
@@ -57,3 +62,156 @@ void ROOT::RDF::RunGraphs(std::vector<RResultHandle> handles)
+  const auto eventsPerTimeInterval = currentEventCount - fLastProcessedEvents;
+  fLastProcessedEvents = currentEventCount;
+
+  const auto oldPrintTime = fLastPrintTime;
+  fLastPrintTime = system_clock::now();
+
+  const duration<double> secondsCurrentInterval = fLastPrintTime - oldPrintTime;
+  fEventsPerSecondStatistics[fEventsPerSecondStatisticsIndex++ % fEventsPerSecondStatistics.size()] = eventsPerTimeInterval / secondsCurrentInterval.count();
+
+  return {currentEventCount, duration_cast<seconds>(fLastPrintTime - fBeginTime)};
+}
+
+namespace {
+  struct PacksOfThree {
+    std::array<unsigned int, 7> packsOfThree; // More not supported by std::size_t
+    int packCounter = 0;
+
+    PacksOfThree(std::size_t count) {
+      for (; count > 0; ++packCounter) {
+        assert(packCounter < static_cast<int>(packsOfThree.size()));
+        packsOfThree[packCounter] = count % 1000;
+        count /= 1000;
+      }
+      --packCounter;
+    }
+  };
+
+  /// Format event counts as `6.346.362k`.
+  std::ostream& operator<<(std::ostream& stream, const PacksOfThree& packs) {
+    for (int i = static_cast<int>(packs.packCounter); i >= 1; --i) {
+      if (i == packs.packCounter) stream << packs.packsOfThree[i];
+      else stream << std::setw(3) << std::setfill('0') << std::right << packs.packsOfThree[i];
+
+      stream << (i > 1 ? '.' : 'k');
+    }
+    return stream << std::setfill(' ');
+  }
+
+  /// Format std::chrono::seconds as `1:30m`.
+  std::ostream& operator<<(std::ostream& stream, std::chrono::seconds elapsedSeconds) {
+    stream << std::chrono::duration_cast<std::chrono::minutes>(elapsedSeconds).count()
+        << ":" << std::setw(2) << std::right << std::setfill('0') << elapsedSeconds.count() % 60;
+    return stream << 'm' << std::setfill(' ');
+  }
+
+  struct RestoreStreamState {
+    RestoreStreamState(std::ostream& stream) :
+      fStream(stream),
+      fFlags(stream.flags()),
+      fFillChar(stream.fill()) { }
+    ~RestoreStreamState() {
+      fStream.setf(fFlags);
+      fStream.fill(fFillChar);
+    }
+
+    std::ostream& fStream;
+    std::ios_base::fmtflags fFlags;
+    std::ostream::char_type fFillChar;
+  };
+}
+
+/// Print event and time statistics.
+void ROOT::RDF::ProgressHelper::PrintStats(std::ostream& stream, std::size_t currentEventCount, std::chrono::seconds elapsedSeconds) const {
+  const auto evtpersec = EvtPerSec();
+  RestoreStreamState restore(stream);
+
+  stream << "[" << elapsedSeconds << "  ";
+
+  // Event counts:
+  if (fUseShellColours) stream << "\e[32m";
+
+  stream << PacksOfThree(currentEventCount);
+  if (fMaxEvents != 0) {
+    stream << "/" << PacksOfThree(fMaxEvents);
+  }
+  stream << " evt  ";
+
+  if (fUseShellColours) stream << "\e[0m";
+
+
+  // events/s
+  stream << std::scientific << std::setprecision(2) << evtpersec << " evt/s";
+
+
+  // Time statistics:
+  if (fMaxEvents != 0) {
+    if (fUseShellColours) stream << "\e[35m";
```

Comment:
This kind of formatting is indeed not going to work on Windows...

Reply:
`#ifdef _WIN32`?

What about `\r` for a carriage return?

Reply:
everything like `"\e[35m"` will not work, so `fUseShellColours` must simply be false on Windows

Reply:
Done.